### PR TITLE
fix(tasks): regenerate recurring tasks with a #tag between the rule and date

### LINKED ISSUE #358: Recurring task with a #tag between the 🔁 rule and the date fails to regenerate
## Problem

Checking off a recurring task does **not** create the next occurrence when a `#tag` sits between the `🔁` recurrence rule and the following date token (`⏳`/`📅`/`➕`).

Real example from `vault/tasks/repeating.md`:

```
- [ ] Check that https://oldergay.men/privacy is up-to-date with services we use 🔼 ➕ 2026-06-15 🔁 every 6 months when done #stage/back-stage ⏳ 2026-07-01
```

Completing this task marks it `[x]` but no new occurrence is generated.

## Root cause

In `POST /task/toggle` (`src/web-server.js:7069`) the recurrence rule is extracted with:

```js
const recurringMatch = taskLine.match(/🔁\s+(.+?)(?:\s+(?:⏳|📅|➕)|$)/);
```

This stops capturing only at `⏳`/`📅`/`➕` — **not** at a `#tag`. So the captured pattern becomes:

```
every 6 months when done #stage/back-stage
```

That string is passed to `calculateNextRecurrence` (`src/web-server.js:6976`), which:

1. Only strips `when done` when it's anchored to the end of the string (`/\s+when\s+done\s*$/i`) — here it's followed by the tag, so nothing is stripped.
2. Matches `/^every\s+...(day|week|month|year...)$/i`, which requires the string to **end** with the unit. It ends with `#stage/back-stage`, so the match fails → returns `null` → no new occurrence.

The interval syntax itself (`every 6 months`) is fully supported — the tag placement is the only trigger. Tasks that keep tags out of the gap between the rule and the date regenerate fine.

The DB-sync parser in `plugins/markdown-tasks/read.js:246` already handles this correctly (it uses `[^\s#]` to stop at tags); the toggle route's regex is just looser.

## Fix

Make the toggle-route extraction stop at `#` as well, matching `read.js` behavior:

```js
const recurringMatch = taskLine.match(/🔁\s+(.+?)(?:\s+(?:⏳|📅|➕|#)|$)/);
```

Add a regression test — `calculateNextRecurrence` and the toggle-route recurrence extraction currently have no test coverage.

### PR DESCRIPTION
**Pull request by jeffcovey**
Fixes #358

## Problem

Checking off a recurring task did not create the next occurrence when a `#tag` sat between the `🔁` rule and the following date token. Real example from `vault/tasks/repeating.md`:

```
- [ ] Check that https://oldergay.men/privacy is up-to-date with services we use 🔼 ➕ 2026-06-15 🔁 every 6 months when done #stage/back-stage ⏳ 2026-07-01
```

Completing it marked the task `[x]` but generated no new occurrence.

## Root cause

`POST /task/toggle` extracted the rule with `/🔁\s+(.+?)(?:\s+(?:⏳|📅|➕)|$)/` (`src/web-server.js`), which stops only at `⏳`/`📅`/`➕` — not at a `#tag`. The captured pattern became `every 6 months when done #stage/back-stage`, which then failed the strict `^every … unit$` match in `calculateNextRecurrence` (the trailing tag also defeated the end-anchored `when done` strip). Result: `null` → no regeneration.

The `every 6 months` syntax itself is fully supported; only the tag placement triggered the bug. The DB-sync parser in `plugins/markdown-tasks/read.js` already handled this correctly via `[^\s#]`.

## Fix

Add `#` to the delimiter set so extraction stops at a tag too:

```js
const recurringMatch = taskLine.match(/🔁\s+(.+?)(?:\s+(?:⏳|📅|➕|#)|$)/);
```

## Verification

- Ran the real `calculateNextRecurrence` + the updated regex from source against the exact reported line: captures `every 6 months when done` → next occurrence `2027-01-10` (previously no match).
- Added a regression test in `test/web-server-tasks.test.js` (toggle a recurring task with a tag between the rule and the date; assert both a completed occurrence and a fresh unchecked one exist). `calculateNextRecurrence` and this path previously had no test coverage.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/web-server.js [hunk 1]
```diff
@@ -7066,7 +7066,10 @@ app.post('/task/toggle', authMiddleware, async (req, res) => {
     debug(`[TASK] Processing task from file: "${taskLine}"`);
 
     // Check if this is a recurring task (has 🔁 pattern)
-    const recurringMatch = taskLine.match(/🔁\s+(.+?)(?:\s+(?:⏳|📅|➕)|$)/);
+    // Stop capturing at the next date token OR a #tag — a tag placed between
+    // the rule and the date would otherwise be swept into the pattern and
+    // break calculateNextRecurrence (matches plugins/markdown-tasks/read.js).
+    const recurringMatch = taskLine.match(/🔁\s+(.+?)(?:\s+(?:⏳|📅|➕|#)|$)/);
```
[the remark this answers is not in the document]
Comment (jeffcovey) [reply src/web-server.js]: Good catch — fixed in 68a12d0e. `🛫` is now included in the terminator set, and I pulled that set into a shared `RECURRENCE_TERMINATOR_TOKENS` constant used by both this route and `/task/edit` so they can't drift again. Added a regression test for the `🔁 every week 🛫 …` case.

File: test/web-server-tasks.test.js [hunk 2]
```diff
@@ -350,6 +352,43 @@ describe('Task Toggle - Edge Cases', () => {
...
+      body: JSON.stringify({
+        filePath: TEST_FILE,
+        lineNumber: RECUR_TAG_LINE,
+        completed: true
+      })
+    });
+
+    expect(response.ok).toBe(true);
+    const result = await response.json();
+    expect(result.updatedLine).toMatch(/- \[x\]/);
```
[the remark this answers is not in the document]
Comment (jeffcovey) [reply test/web-server-tasks.test.js:379]: Done in 68a12d0e — both recurrence regression tests now assert `result.success` alongside the file-content checks.

File: src/web-server.js [hunk 3]
```diff
@@ -7221,7 +7233,9 @@ app.post('/task/edit', authMiddleware, async (req, res) => {
     const priorityEmoji = priority && priorityEmojis[priority] ? priorityEmojis[priority] : '';
 
     // Preserve recurrence pattern from original line (stop at next date emoji or end)
-    const recurringMatch = taskLine.match(/🔁\s+[^⏳📅🛫✅➕\n]+/);
+    const recurringMatch = taskLine.match(
+      new RegExp(`🔁\\s+[^${RECURRENCE_TERMINATOR_TOKENS.join('')}\\n]+`)
+    );
     const recurrence = recurringMatch ? recurringMatch[0].trim() : '';
```
[the remark this answers is not in the document]
Comment (jeffcovey) [reply src/web-server.js]: @copilot please fix this. Use the same alternation-based terminator as `/task/toggle` (reuse `RECURRENCE_TERMINATOR_TOKENS`) and include `#` as a terminator here too, so the preserved recurrence no longer swallows a trailing `#tag` (avoiding the duplicate-tag reconstruction) and avoids the surrogate-pair character-class hazard. Please also add/extend a test covering editing a recurring task that has a `#tag` between the 🔁 rule and the date.
